refactor(KL-147/API 요청): improve structure of Components

seoulyego opened the pull request:
## 📌 연관된 이슈
[KL-147/API 요청](https://ohhamma.atlassian.net/browse/KL-147)

## 📝 작업 내용
- PreviewContent의 크기를 고정폭 기준으로 변경하였습니다.
- antd 컴포넌트의 디자인과 테마를 수정하였습니다.
  - ConfigProvider 적극 활용하여 디자인 코드 스타일을 통일하였습니다.
  - 태그, 초기화, 정렬 버튼을 수정하였습니다
  - Pagination의 API를 수정하였습니다.
- DataFetcher의 동작과 책임을 명확하고 단순하게 변경하였습니다.
  - ProductDataFetcher로부터 early return UI를 분리하였습니다.
  - TagSwitchList의 구조를 ProductList와 동일하게 수정하였습니다.
  - TagSwitchList의 Fetching 로직을 TagDataFetcher에 위임하였습니다.
  - TagDataFetcher의 기존 동작은 훅으로 분리하였습니다.
- BasicFilter의 구조를 개선하였습니다.
  - Container와 Component의 경계를 명확히 구분하였습니다.
  - Component의 명명 규칙을 통일하였습니다.
  - RegionCollapse의 Fetching 로직을 RegionArray로 분리하였습니다.

## 🌳 작업 브랜치명
KL-147/ApiRequest

## 📸 스크린샷 (선택)

## 💬 리뷰 요구사항 (선택)
> 리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit


- **New Features**
	- Introduced a new `ProductDataStatusRenderer` component to manage loading and error states for product data.
	- Added a `TagDataStatusRenderer` component for improved tag data status handling.

- **Bug Fixes**
	- Improved loading and error state management in the FeedPage for better user experience.

- **Style**
	- Updated styling for multiple components to enhance layout consistency and visual coherence.

- **Refactor**
	- Streamlined component structures and logic for improved readability and maintainability.
	- Simplified pagination and filtering logic across various components.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

seoulyego committed:
KL-147/design: for antd button

- 모든 PreviewContent의 크기가 동일하게 변경되었습니다.
	- width를 고정 크기로 변경하였습니다.
	- tagContainer의 높이를 고정하였습니다.
- antd 버튼들의 테마를 변경하였습니다.
	- antd의 ConfigProvider를 '적극' 활용하여 theme를 설정하였습니다.
	- 초기화 버튼과 태그 버튼, 정렬 버튼의 font가 변경되었습니다.
- Pagination의 일부 수정되었습니다.

seoulyego committed:
KL-147/refactor: separate early return UI from ProductDataFetcher

- ProductDataFetcher로부터 early return UI를 분리하였습니다.
- ProductList의 props 명을 more common name으로 변경하였습니다.

seoulyego committed:
KL-147/refactor: separate fetching logic from TagSwitchList

- ProductList의 파일 구조와 유사하게 변경하였습니다.
- TagSwitchList로부터 fetching 작업을 분리하였습니다.
	- 분리된 fetching 작업이 새로운 TagDataFetcher가 되었습니다.
	- TagDataFetcher의 기존 동작은 훅으로 분리하였습니다.

seoulyego committed:
KL-147/refactor: improve structure of BasicFilter

- BasicFilter의 구조를 전반적으로 정리하고 개선하였습니다.
	- Container와 Component의 경계를 명확히 구분하였습니다.
	- Component의 명명 규칙을 통일하였습니다.
	- RegionCollapse의 fetching 동작을 RegionArray로 분리하였습니다.

seoulyego committed:
KL-147/rename: rename SubCategoryContainer to SubCategoryArray

SubCategoryContainer 컴포넌트의 이름이 변경됨에 따라 파일명도 동일하게 변경하였습니다.
- SubCategoryArray의 import 경로를 수정하였습니다.

sayoonnn commented on src/pages/FeedPage/components/ProductList/ProductDataFetcher.jsx:
```diff
@@ -66,15 +64,7 @@ function ProductDataFetcher({ children }) {
-        <StyledList className="empty">로딩에 실패했습니다.</StyledList>
-      )}
-      {!isLoading && !isError && children({ productDataList })}
+      {children({ isLoading, productDataList, isError })}
```
FeedPage에서는 어떤 컴포넌트가 오는지 확인하긴 편한데,
ProductPatcher에서는 오히려 가독성이 떨어지는 것 같아요 갑자기 children 함수를 호출하면
여기에 오는 게 컴포넌트인지, 단순히 data를 파싱해서 값을 붙여넣은 것인지 햇갈립니다

또 fetching 하는 부분과 render 하는 부분을 분리하는 게 좋다고도 하셨는데 pagination 은 그냥 render 해버리는 것도 안맞는 것 같습니다

seoulyego replied:
1. ProductList는 FeedPage 내의 여타 컴포넌트보다 가장 핵심적인 기능을 수행하는 UI이기 때문에 
ProductList라는 UI로 렌더된다는 것을 FeedPage에서 보여주는 것이 필요하다고 생각했습니다. 
그렇기 때문에 DataFetcher에서는 fetching에만 집중하도록 하고,
fetching status와 data를 children이라는 함수를 통해 전달하여 status에 맞는 UI를 렌더할 수 있도록 되어 있습니다.
children이 함수인 이유는 위 조건을 만족하면서 하위 컴포넌트에 props를 전달할 수 있도록 하기 위함으로 이해해주시면 좋겠습니다.

2. Pagination이 모든 fetching status에서 필요하기에 해당 위치에서 사용하고 있었습니다.
여러 설명을 작성하고 있었으나 render 로직의 컴포넌트를 별도로 만들면 이해하기에 좀 더 편할 것 같아서 Renderer라는 컴포넌트를 새로 작성하여 분리하였습니다.

sayoonnn commented on src/pages/FeedPage/components/BasicFilter/CountrySection.jsx:
```diff
@@ -82,113 +79,92 @@ function CountryRadioArray({ regionId }) {
-  const [isOpen, setIsOpen] = useState({})
   const { isLoading, data: regionsData, isError } = useKyQuery('regions')
 
   useEffect(() => {
```
regionsData.data를 다시 state에 넣는 이유가 궁금합니다
regionsData.data.map으로 값을 반복해서 넣으면 결국에 regionsData 랑 regions는 같은 값을 저장하는 게아닌가요..?

불필요한 state 관리를 없애려고 react-query를 사용해서 메모리에  fetching 한 값을 저장하는데
이렇게 되면 두번 저장하는  것 같습니다

seoulyego replied:
변경하여 새로운 커밋으로 올렸습니다

sayoonnn reviewed (commented):
작성한 부분 한 번 확인해주시면 감사하겠습니당

seoulyego committed:
KL-147/refactor: separate UI branching logic based on fetching status

fetching status에 따른 UI render 분기 로직을 Renderer 컴포넌트로 분리하였습니다.

seoulyego committed:
KL-147/fix: fix to infinite rerendering

무한 재렌더링 문제를 해결하였습니다.

seoulyego committed:
KL-147/refactor: delete states to store fetching data

fetching된 데이터를 저장하는 상태를 제거하였습니다.

seoulyego committed:
KL-147/refactor: modify to the same structure as the changed ProductList

변경된 ProductList와 동일한 구조로 수정하였습니다.
	- fetching data를 저장하는 상태를 제거하였습니다.
	- fetching status에 따른 UI render 분기 로직을 Render 컴포넌트로 분리하였습니다.

sayoonnn reviewed (approved):
확인했습니당
